lightwood/encoder/categorical/autoencoder.py

```diff
@@ -11,7 +11,7 @@
 
 
 class CategoricalAutoEncoder(BaseEncoder):
-    def __init__(self, stop_after: int = 3600, is_target=False, max_encoded_length=100):
+    def __init__(self, stop_after: int = 3600, is_target=False, max_encoded_length=100, use_autoencoder=None):
```

Comment:
Make the default be a string called `'auto'` instead of `None` so it makes more sense.

Provide type for this and the rest of the args please